redbot/cogs/warnings/warnings.py

```diff
@@ -57,10 +58,31 @@ async def red_delete_data_for_user(
 
         all_members = await self.config.all_members()
 
-        async for guild_id, guild_data in AsyncIter(all_members.items(), steps=100):
-            if user_id in guild_data:
+        c = 0
+
+        for guild_id, guild_data in all_members.items():
+            c += 1
+            if not c % 100:
+                await asyncio.sleep(0)
+
+            if str(user_id) in guild_data:
                 await self.config.member_from_ids(guild_id, user_id).clear()
 
+            for remaining_user, user_warns in guild_data.items():
+                c += 1
+                if not c % 100:
+                    await asyncio.sleep(0)
+
+                for warn_id, warning in user_warns.get("warnings", {}):
+                    c += 1
+                    if not c % 100:
+                        await asyncio.sleep(0)
+
+                        if warning.get("mod", 0) == user_id:
+                            await self.config.member_from_ids(guild_id, remaining_user).set_raw(
+                                "warnings", warn_id, "mod", value=0xDE1
```

Comment:
Could do special handling (i.e. not showing ID for deleted user) for this case in `[p]warnings` and `[p]mywarnings`, but this is not critical.

Reply:
compared to the other changes, this one's a small deal, but I'll look at it in a sec